### PR TITLE
feat: pause contract

### PR DESCRIPTION
**Pull request by dongcool**

### CODE REVIEW

File: contracts/linear/src/owner.rs [hunk 1]
```diff
@@ -98,7 +106,20 @@ impl LiquidStakingContract {
...
     pub fn configure_liquidity_pool(&mut self, config: LiquidityPoolConfig) {
+        self.assert_running();
         self.assert_owner();
         self.liquidity_pool.configure(config);
     }
+
+    // --- Pause ---
+
+    pub fn pause(&mut self) {
+        self.assert_owner();
```
Comment (linguists) [inline contracts/linear/src/owner.rs:117]: may assert contract is not paused:

```rust
require!(!self.paused, "The contract is already paused")
```

File: contracts/linear/src/owner.rs [hunk 2]
```diff
@@ -98,7 +106,20 @@ impl LiquidStakingContract {
...
+        self.paused = true;
+    }
+
+    pub fn resume(&mut self) {
+        self.assert_owner();
```
Comment (linguists) [inline contracts/linear/src/owner.rs:122]: may assert contract has been paused:

```rust
require!(self.paused, "The contract is not paused")
```

**Review by linguists — CHANGES_REQUESTED** [review]

File: contracts/linear/src/errors.rs [hunk 3]
```diff
@@ -7,6 +7,7 @@ pub const ERR_NO_ENOUGH_INIT_DEPOSIT: &str =
 
 // owner
 pub const ERR_NOT_OWNER: &str = "Only owner can perform this action";
+pub const ERR_PAUSED: &str = "The contract is paused now, please try later";
```
Comment (linguists) [inline contracts/linear/src/errors.rs:10]:
```suggestion
pub const ERR_PAUSED: &str = "The contract is paused now. Please try later";
```

**Review by linguists — CHANGES_REQUESTED** [review]

File: tests/__tests__/linear/pause.ava.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    test,
+    transfer(contract, alice, owner, transferAmount),
+    'The contract is paused now, please try later'
+  );
+
+  await assertFailure(
+    test,
+    transferCall(contract, alice, owner, transferAmount, ''),
+    'The contract is paused now, please try later'
+  );
```
Comment (linguists) [inline tests/__tests__/linear/pause.ava.ts:99]: Should we also test `stake` and `unstake` here?

**Review by 0xdefacto — APPROVED** [review]

**Pull request merged**